File: algorithms/equal_stacks/kevin.py
#!/usr/bin/env python3
# Kevin Boyette


# h1 = [3, 2, 1, 1, 1]
# h2 = [4, 3, 2]
# h3 = [1, 1, 4, 1]

n1, n2, n3 = input().strip().split(' ')
n1, n2, n3 = map(int, [n1, n2, n3])
h1 = [int(h1_temp) for h1_temp in input().strip().split(' ')]
h2 = [int(h2_temp) for h2_temp in input().strip().split(' ')]
h3 = [int(h3_temp) for h3_temp in input().strip().split(' ')]


def first_is_biggest(x, y, z): return x > y or x > z


def all_equal(x, y, z): return x == y and y == z

h1_total = sum(h1)
h2_total = sum(h2)
h3_total = sum(h3)

equal_height = False

while not equal_height:
    if first_is_biggest(h1_total, h2_total, h3_total):
        h1_total -= h1.pop(0)
    if first_is_biggest(h2_total, h1_total, h3_total):
        h2_total -= h2.pop(0)
    if first_is_biggest(h3_total, h1_total, h2_total):
        h3_total -= h3.pop(0)
    if all_equal(h1_total, h2_total, h3_total):
        equal_height = True

print(h1_total)

File: algorithms/equal_stacks/kevin.js
'use strict'
const kevin = () => {
  console.log('HackerRank Javascript Attempt By Kevin Boyette')
}
kevin()
const equalStacks = () => {
  const h1 = [3, 2, 1, 1, 1]
  const h2 = [4, 3, 2]
  const h3 = [1, 1, 4, 1]

  let equalHeight = false
  const sum = (total, x) => { return total + x }
  const firstParamIsBiggest = (x, y, z) => { return x > y || x > z }
  const allAreEqual = (x, y, z) => { return x === y && y === z }

  let h1Total = h1.reduce(sum)
  let h2Total = h2.reduce(sum)
  let h3Total = h3.reduce(sum)

  while (!equalHeight) {
    if (firstParamIsBiggest(h1Total, h2Total, h3Total)) { h1Total -= h1.shift() }
    if (firstParamIsBiggest(h2Total, h1Total, h3Total)) { h2Total -= h2.shift() }
    if (firstParamIsBiggest(h3Total, h1Total, h2Total)) { h3Total -= h3.shift() }
    if (allAreEqual(h1Total, h2Total, h3Total)) { equalHeight = true }
  }
  console.log(h1Total)
}
equalStacks()
